### PR TITLE
Setup Github Actions

### PR DESCRIPTION
**Pull request by Egorand**
Largely copied from https://github.com/uber/RxDogTag/blob/master/.github/workflows/ci.yml

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  build:
+    name: JDK ${{ matrix.java_version }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java_version: [1.8, 11]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Install JDK
```
Comment (ZacSweers) [inline .github/workflows/ci.yml]: GitHub fixed a bug that prevented using replacements in names, so you could do `Install JDK ${{ matrix.java_version }}` here if you want. Haven't verified it works yet but someone else on the community forum confirmed it

**Comment by ZacSweers** [comment]
Fun thing to do on these - compare how fast the build was on github vs travis :)

image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Egorand) [reply .github/workflows/ci.yml]: Oh cool, yeah, I'll do it!

**Comment by Egorand** [comment]
> compare how fast the build was on github vs travis
Totally, I'm very hyped about it!

**Comment by Egorand** [comment]
One thing I'm not sure about is whether we need both push and pr, cause they're different - while push simply builds the branch, pr rebases on the latest master and builds, which can uncover different types of issues. @ZacSweers do you know if there's a way to declare both in YAML without duplicating steps setup?

**Comment by ZacSweers** [comment]
Didn't realize pr rebases against latest master - that's neat! Unfortunately I don't think there's an easy way to share them unfortunately. Been a month or so since I last tried though

**Pull request merged**